doc/source/development.rst

```diff
@@ -193,143 +201,242 @@ Make and commit your changes.
+
+
+Documentation
+~~~~~~~~~~~~~
+
+KIWI uses `Sphinx <https://www.sphinx-doc.org/en/master/>`_ for the API and
+user documentation.
+
+In order to build the html documentation call:
+
+.. code-block:: shell-session
+
+    tox -e doc
+
+or to build the full documentation (including a PDF generated by LaTeX
+[#f3]_):
+
+.. code-block:: shell-session
+
+    tox -e packagedoc
+
+
+Document all your classes, methods, their parameters and their types using
+the standard `reStructuredText
+<https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html>`_
+syntax as supported by Sphinx, an example class is documented as follows:
+
+.. code-block:: python
+
+   class Example(object):
+       """
+       **Example class**
+
+       :param str param: A parameter
+       :param bool : Source file name to compress
+       :param list supported_zipper: List of supported compression tools
+       :attr Optional[str] attr: A class attribute
+       """
+       def __init__(self, param, param_w_default=False):
+           self.attr = param if param_w_default else None
+
+       def method(self, param):
+           """
+           A method that takes a parameter.
+
+           :param list param: a parameter
+           :return: whether param is very long
+           :rtype: bool
+           """
+           return len(parame) > 50
+
+
+Try to stick to the following guidelines when documenting source code:
+
+- Classes should be documented directly in their main docstring and not in
+  `__init__`.
+
+- Document **every** parameter including their types.
+
+- Only public methods should be documented, private methods don't have to,
+  unless they are complex and it is not easy to grasp what they do (which
+  should be avoided anyway).
+
+
+Please also document any user-facing changes that you implementing
+(e.g. adding a new build type) in the user documentation, which can be
+found in `doc/source`. General documentation should be put into the
+`working_with_kiwi/` subfolder, whereas documentation about more
+specialized topics would belong into the `building/` subfolder.
+
+Please adhere to a line limit of 75 characters when writing the user facing
+documentation [#f2]_.
+
+
+Additional Information
+----------------------
+
+The following sections provides further information about the repository
+integrity, version, package and documentation management.
 
 .. _Signing_Git_Patches:
 
 Signing Git Patches
 ~~~~~~~~~~~~~~~~~~~
 
-With ssh keys being widely available and the increasing compute power
-available to many people refactoring of SSH keys is in the range of
-possibilities. Therefore SSH keys as used by GitHub as a
-"login/authentication" mechanism no longer provide the security they
-once did. In an effort to ensure the integrity of the repository and
-the code base patches sent for inclusion must be GPG signed.
+To ensure the integrity of the repository and the code base, patches sent
+for inclusion should be signed with a GPG key.
```

Comment:
Sorry to nitpick here, but the small word "should" kind of contradicts the whole integrity part.

I think, you need to be very clear what the project wants. Is it ok to accept patches which are not signed by a GPG key? Or are signed patches only a recommendation? Or do you *only* accept signed patches?

With *should* you indicate signing patches are optional. Was that your intention?

Reply:
cc @schaefi 

We **want** to have commits GPG signed, but have no mechanism in place that actually enforces this, thus I've changed the wording to describe the current state.